Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41030&attempt_id=0dd50254-f81d-4d79-8b89-c16d05acb45e)

Attempted: nheko

<details><summary>Partial log (click to expand)</summary><p>

```
[ 69%] Building CXX object CMakeFiles/nheko.dir/include/dialogs/moc_JoinRoom.cpp.o
[ 69%] Building CXX object CMakeFiles/nheko.dir/include/dialogs/moc_MemberList.cpp.o
[ 70%] Building CXX object CMakeFiles/nheko.dir/include/dialogs/moc_LeaveRoom.cpp.o
20 errors generated.
make[2]: *** [CMakeFiles/nheko.dir/build.make:2023: CMakeFiles/nheko.dir/src/notifications/ManagerMac.mm.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:68: CMakeFiles/nheko.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
builder for '/nix/store/l8x5f9dyr50srn3h8czrihcn2jlarffv-nheko-0.4.1.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/l8x5f9dyr50srn3h8czrihcn2jlarffv-nheko-0.4.1.drv' failed
```
</p></details>

--- a/pkgs/applications/networking/instant-messengers/nheko/default.nix
+++ b/pkgs/applications/networking/instant-messengers/nheko/default.nix
@@ -1,4 +1,7 @@
-{ stdenv, fetchFromGitHub, fetchurl, cmake, doxygen, lmdb, qt5 }:
+{
+  lib, stdenv, fetchFromGitHub, fetchurl,
+  cmake, doxygen, lmdb, qt5, qtmacextras
+}:
 
 let
   json_hpp = fetchurl {
@@ -60,6 +63,22 @@ stdenv.mkDerivation rec {
   # a dependency is updated, so that the fetches up there can be updated too
   patches = [ ./external-deps.patch ];
 
+  # If, on Darwin, you encounter the error
+  #   error: must specify at least one argument for '...' parameter of variadic
+  #   macro [-Werror,-Wgnu-zero-variadic-macro-arguments]
+  # Then adding this parameter is likely the fix you want.
+  #
+  # However, it looks like either cmake doesn't honor this CFLAGS variable, or
+  # darwin's compiler doesn't have the same syntax as gcc for turning off
+  # -Werror selectively.
+  #
+  # Anyway, this is something that will have to be debugged with access to a
+  # darwin-based OS. Sorry about that!
+  #
+  #preConfigure = lib.optionalString stdenv.isDarwin ''
+  #  export CFLAGS=-Wno-error=gnu-zero-variadic-macro-arguments
+  #'';
+
   cmakeFlags = [
     "-DMATRIX_STRUCTS_LIBRARY=${matrix-structs}/lib/static/libmatrix_structs.a"
     "-DMATRIX_STRUCTS_INCLUDE_DIR=${matrix-structs}/include/matrix_structs"
@@ -71,7 +90,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     lmdb lmdbxx matrix-structs qt5.qtbase qt5.qtmultimedia qt5.qttools tweeny
-  ];
+  ] ++ lib.optional stdenv.isDarwin qtmacextras;
 
   enableParallelBuilding = true;
 

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -17247,7 +17247,7 @@ with pkgs;
 
   nedit = callPackage ../applications/editors/nedit { };
 
-  nheko = callPackage ../applications/networking/instant-messengers/nheko { };
+  nheko = libsForQt5.callPackage ../applications/networking/instant-messengers/nheko { };
 
   nomacs = libsForQt5.callPackage ../applications/graphics/nomacs { };
 

